packages/client-utils/test/sync-tools.test.js

```diff
@@ -13,13 +14,43 @@ import {
 const retryIntervalMs = 10;
 const DEFAULT_TIMEOUT = 30;
 
+const makeSimpleMarshaller = () => {
+  const vals = [];
+  const fromVal = val => {
+    vals.push(val);
+    return vals.length - 1;
+  };
+  const toVal = slot => vals[slot];
+  return makeMarshal(fromVal, toVal, {
+    serializeBodyFormat: 'smallcaps',
+    marshalSaveError: err => {
+      throw err;
+    },
+  });
+};
+harden(makeSimpleMarshaller);
+
 const makeFakeFollow = () => {
-  let value = [[]];
+  const marshaller = makeSimpleMarshaller();
+  let value;
 
   const setValue = newValue => (value = newValue);
   const follow = () => Promise.resolve(value);
+  /**
+   *
+   * @param {any} _
+   * @param {string} path Assumes the path will be something like 'published.auction.book0'
+   * where value = { book0: {...} }
+   */
+  const followByPath = (_, path) => {
+    const propName = path.split('.').at(-1);
+    return Promise.resolve(
+      // @ts-expect-error path will be a sequence of strings joined by "."
+      JSON.stringify(marshaller.toCapData(value[propName])),
+    );
+  };
 
-  return { setValue, follow };
+  return { setValue, follow, followByPath, marshaller };
```

Comment:
please use existing capabilities

Reply:
Hmm... I found it helpful to use those for mocking the vstorage in the tests. But that was before we had `client-tools` I'll try to come with an approach that mock vstorageKit as well.

Reply:
oh sorry, this is a test. do whatever you want in the test

Reply:
Did something in https://github.com/Agoric/agoric-sdk/pull/10385/commits/1594fd400e86246d43bc9a4bd655a49ea2998ed8.